web/src/actions/admin.js

```diff
@@ -0,0 +1,44 @@
+import { notify } from './notification';
+import axios from '../util/api';
+
+export const ADMIN_CREATE_USER_REQUEST = Symbol(
+  'admin : create user : request'
+);
+export const ADMIN_CREATE_USER_SUCCESS = Symbol(
+  'admin : create user : success'
+);
+export const ADMIN_CREATE_USER_ERROR = Symbol('admin : create user : error');
+
+export const createUser = ({ email, name, password, state }) => dispatch => {
+  dispatch({ type: ADMIN_CREATE_USER_REQUEST });
+
+  return axios
+    .post('/users', { email, name, password, state })
+    .then(() => {
+      dispatch({ type: ADMIN_CREATE_USER_SUCCESS });
+      dispatch(notify('User created!'));
+    })
+    .catch(e => {
+      dispatch({ type: ADMIN_CREATE_USER_ERROR });
+
+      switch (e.response.data.error) {
```

Comment:
This is one of the clearest patterns for handling this sort of interaction I've seen. :highfive: